crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -20,6 +21,77 @@ where
         unimplemented!()
     }
 
+    /// Finds a given trasaction by its hash.
+    pub(crate) async fn transaction_by_hash(
+        &self,
+        hash: H256,
+    ) -> EthResult<Option<reth_rpc_types::Transaction>> {
+        let tx_by_hash = self.client().transaction_by_hash(hash)
+            .map_err(|_| EthApiError::EmptyRawTransactionData)?
+            .ok_or(EthApiError::EmptyRawTransactionData)?
```

Comment:
I don't think this is correct, because this function is supposed to return null if the transaction does not exist, right?